packages/angular/ssr/src/i18n.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/**
+ * Extracts a potential locale ID from a given URL based on the specified base path.
+ *
+ * This function parses the URL to locate a potential locale identifier that immediately
+ * follows the base path segment in the URL's pathname. If the URL does not contain a valid
+ * locale ID, an empty string is returned.
+ *
+ * @param url - The full URL from which to extract the locale ID.
+ * @param basePath - The base path used as the reference point for extracting the locale ID.
+ * @returns The extracted locale ID if present, or an empty string if no valid locale ID is found.
+ *
+ * @example
+ * ```js
+ * const url = new URL('https://example.com/base/en/page');
```

Comment:
Quick question: is this the format of representing locale info that we have today (and we rely on) as well or it's a new format?

Reply:
This is the existing format.